packages/imperative/src/config/src/ProfileInfo.ts

```diff
@@ -1469,11 +1455,58 @@ export class ProfileInfo {
                 from: [typeInfo.sourceApp],
                 latestFrom: typeInfo.schema.version ? typeInfo.sourceApp : undefined
             };
-            this.updateSchemaAtLayer(profileType, typeInfo.schema);
+            addedToSchema = this.updateSchemaAtLayer(profileType, typeInfo.schema, layer);
+        }
+
+        return { success: addedToSchema, info: infoMsg };
+    }
+
+    /**
+     * Adds a profile type to the schema, and tracks its contribution in extenders.json.
+     *
+     * NOTE: `readProfilesFromDisk` must be called at least once before adding new profile types.
+     *
+     * @param {string} profileType The new profile type to add to the schema
+     * @param {IExtenderTypeInfo} typeInfo Type metadata for the profile type (schema, source app., optional version)
+     * @param [updateProjectSchema] Automatically update a project-level schema if one exists.
+     * @returns {IAddProfTypeResult} the result of adding the profile type to the schema
+     */
+    public addProfileTypeToSchema(profileType: string, typeInfo: IExtenderTypeInfo, updateProjectSchema?: boolean): IAddProfTypeResult {
+        // Get the active team config layer
+        const activeLayer = this.getTeamConfig()?.layerActive();
+        if (activeLayer == null) {
+            return {
+                success: false,
+                info: "This function only supports team configurations."
+            };
+        }
+
+        // copy last value for `extenders.json` to compare against updated object
+        const oldExtendersJson = lodash.cloneDeep(this.mExtendersJson);
+        let result: IAddProfTypeResult = {
+            success: false,
+            info: ""
+        };
+
+        let wasGlobalUpdated = false;
+
+        if (!activeLayer.global) {
+            if (updateProjectSchema) {
+                result.success = this.updateSchemaAtLayer(profileType, typeInfo.schema, activeLayer);
+            }
+            wasGlobalUpdated = this.addToGlobalSchema(profileType, typeInfo).success;
+        } else {
+            result = this.addToGlobalSchema(profileType, typeInfo);
+            wasGlobalUpdated = result.success;
+        }
+
+        // If we could not add to the schema, return early
+        if (!result.success && !wasGlobalUpdated) {
+            return result;
```

Comment:
Could you help me understand why we need to return early?

it would seems that if we can't update the global layer's schema (i.e. `wasGlobalUpdated === false`) we won't attempt to update the extenders.json file 🐶

Reply:
> it would seem that if we can't update the global layer's schema (i.e. wasGlobalUpdated === false), we won't attempt to update the extenders.json file 🐶

yep, that's correct - if `wasGlobalUpdated` is false, it means one of two scenarios occurred:
- the layer does not exist
- the layer exists, but the schema does not exist on disk

In both cases, we don't want to update `extenders.json` because the schema was left unchanged, and `extenders.json` represents the state of the global schema. For the other scenarios, `wasGlobalUpdated` will be set to `true` and it will then update `extenders.json` as expected 😋 (but only if the contents of `extender.json` have changed)

Reply:
Ah, I'm being dumb - I made a truth table to verify the behavior, and confirmed that I can remove this check since it will have the same effect without it. I'll remove it. 😋 Thanks for catching that!!

Reply:
Resolved in 5200c18